app/src/common/shared/com/igalia/wolvic/browser/api/WSession.java

```diff
@@ -465,13 +465,22 @@ WResult<WAllowOrDeny> onSubframeLoadRequest(
             return null;
         }
 
+        abstract class OnNewSessionRunnable implements Runnable {
```

Comment:
Do I understand correctly that we have setSession() here only to supply it to run()? In this case I think we don't need to extend runnable and instead we can create a custom callback and supply session as an argument.

Reply:
Yeah you're totally right. I'll change that.